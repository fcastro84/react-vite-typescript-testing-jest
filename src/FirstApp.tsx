

const FirstApp = ({ title, subTitle= 'Testing with Jest', name = 'React 18'}: {
    title: string,
    subTitle?: string,
    name?: string
}) => {
  return (
    <>
      <h1 data-testid="test-title">{ title }</h1>
      <p>{ subTitle }</p>
      <p>{ name }</p>
    </>
  )
}

export default FirstApp
