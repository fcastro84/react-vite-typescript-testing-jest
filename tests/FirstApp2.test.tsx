import { render, screen } from "@testing-library/react"
import FirstApp from '../src/FirstApp';
import React from "react";

describe(' Test in component <FirstApp/>', () => {

    const title = 'Hola, soy Goku'
    const subTitle = 'Testing New'

    test('This has make to match with the snapshot ', () => {
      
        
        const { container } = render( <FirstApp title={title} /> )

        expect( container ).toMatchSnapshot()

    })

    test('This has what view of title in h1', () => {
      
        render( <FirstApp title={title} /> )

        expect( screen.getByText(title) ).toBeTruthy()

        expect( screen.getByTestId('test-title').innerHTML ).toContain(title)

        expect( screen.getByRole('heading', { level: 1}).innerHTML).toContain(title)  

    })

    test('This has what view of subTitle sending by props', () => {
      
        render( 
                 <FirstApp 
                        title={title} 
                        subTitle={subTitle}
                 /> 
                )

        expect( screen.getByText(subTitle) ).toBeTruthy()

        expect( screen.getAllByText(subTitle).length ).toBe(1)

        

        

    })
    
})