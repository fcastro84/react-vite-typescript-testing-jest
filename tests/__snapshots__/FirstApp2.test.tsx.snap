// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Test in component <FirstApp/> This has make to match with the snapshot  1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku
  </h1>
  <p>
    Testing with Jest
  </p>
  <p>
    React 18
  </p>
</div>
`;
